### ISSUE TITLE
Continuous testing does not work when Quarkus is run from the IDE

### ISSUE THREAD

**Issue by stuartwdouglas** [quarkusio/quarkus#34152]
### Describe the bug

Parent first test scoped dependencies are not added to the class path when run from the IDE. This results in JUnit being loaded from the wrong class loader.

**Comment by geoand** [comment]
Just to add some extra points to what Stuart said (and to spare anyone debugging in the future), here is what happens:

The JUnit classes are properly attempted by Quarkus to be loaded via the Augmentation ClassLoader which uses the  JVM System ClassLoader as it's parent, so normally we would expect the class to be loaded by it. 
However, as a result of the IDE not knowing that it needs to have JUnit on the runtime classpath (it only thinks it's needed for tests), the JVM System ClassLoader does not include any JUnit dependencies and the JUnit classes end up being loading by the Augmentation ClassLoader as a [fallback](https://github.com/quarkusio/quarkus/blob/82cb5edc3a65cf190cf9c7aa224495a2ab5d486e/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java#L493-L497).

In short, I don't see any good way of having this work...

cc @maxandersen
